File: Binary-Tree/Inoderpreorderpostorder/inprepost.py
class Node:
    def __init__(self, data):
        self.data = data  # Node data
        self.left = None  # Pointer to the left child
        self.right = None # Pointer to the right child

class InPrePost:
    def __init__(self):
        self.root = None  # Initialize root as None

    # Function to insert a new node
    def insert(self, node, data):
        if node is None:
            return Node(data)
        if data < node.data:
            node.left = self.insert(node.left, data)
        else:
            node.right = self.insert(node.right, data)
        return node

    # Preorder traversal
    def preorder(self, node):
        if node is None:
            return
        print(node.data, end=" ")
        self.preorder(node.left)
        self.preorder(node.right)

    # Inorder traversal
    def inorder(self, node):
        if node is None:
            return
        self.inorder(node.left)
        print(node.data, end=" ")
        self.inorder(node.right)

    # Postorder traversal
    def postorder(self, node):
        if node is None:
            return
        self.postorder(node.left)
        self.postorder(node.right)
        print(node.data, end=" ")

def main():
    tree = InPrePost()
    tree.root = tree.insert(tree.root, 1)
    tree.root = tree.insert(tree.root, 2)
    tree.root = tree.insert(tree.root, 3)
    tree.root = tree.insert(tree.root, 4)
    tree.root = tree.insert(tree.root, 5)

    print("Preorder traversal: ", end="")
    tree.preorder(tree.root)
    print()

    print("Inorder traversal: ", end="")
    tree.inorder(tree.root)
    print()

    print("Postorder traversal: ", end="")
    tree.postorder(tree.root)
    print()

if __name__ == "__main__":
    main()

File: Binary-Tree/Inoderpreorderpostorder/inprepost.java
class Node {
    int data;
    Node left, right;
    Node(int data) {
        this.data = data;
        left = right = null;
    }
}
class InPrePost {
    Node root;
    public Node insert(Node node, int data) {
        if (node == null) {
            return new Node(data);
        }
        if (data < node.data) {
            node.left = insert(node.left, data);
        } else {
            node.right = insert(node.right, data);
        }
        return node;
    }
    public void preorder(Node node) {
        if (node == null) {
            return;
        }
        System.out.println(node.data);
        preorder(node.left);
        preorder(node.right);
    }
    public void inorder(Node node) {
        if (node == null) {
            return;
        }
        inorder(node.left);
        System.out.println(node.data);
        inorder(node.right);
    }
    public void postorder(Node node) {
        if (node == null) {
            return;
        }
        postorder(node.left);
        postorder(node.right);
        System.out.println(node.data);
    }

    public static void main(String[] args) {
        InPrePost tree = new InPrePost();
        tree.root = tree.insert(tree.root, 1);
        tree.root = tree.insert(tree.root, 2);
        tree.root = tree.insert(tree.root, 3);
        tree.root = tree.insert(tree.root, 4);
        tree.root = tree.insert(tree.root, 5);
        System.out.println("Preorder traversal: ");
        tree.preorder(tree.root);
    }

}
